easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -918,6 +932,33 @@ def resolve_template(value, tmpl_dict):
 
     return value
 
+def replace_templates(value, templ_const, templ_val):
+    """
+    Given a value, try to substitute template strings where possible.
+        - value can be a string, list, tuple, dict or combination thereof
+        - templ_const is a dictionary of template strings (constants)
+        - templ_val is a dictionary of template strings specific for this easyconfig file
+    """
+    if isinstance(value, basestring):
+        old_value = ""
+        while value != old_value:
+            old_value = value
+            if value in templ_const:
+                value = templ_const[value]
+            else:
+                # check for template values - longest strings first
+                for v in sorted(templ_val, key=lambda v: len(v), reverse=True):
```

Comment:
you're sorting `templ_val` over and over again here, but it doesn't change over time

at least move the sorting bit out of the while loop, and maybe even out of `replace_templates` into `dump`? sorting it once should suffice ;)